packages/host/tests/acceptance/ai-assistant-test.gts

```diff
@@ -0,0 +1,306 @@
+        },
+      },
+    });
+  });
+
+  test('attaches a card in a coversation multiple times', async function (assert) {
+    await visitOperatorMode({
+      stacks: [
+        [
+          {
+            id: `${testRealmURL}index`,
+            format: 'isolated',
+          },
+        ],
+      ],
+    });
+    await click('[data-test-open-ai-assistant]');
+    const testCard = `${testRealmURL}Person/hassan`;
+
+    for (let i = 1; i <= 3; i++) {
+      await waitFor('[data-test-message-field]');
+      await fillIn('[data-test-message-field]', `Message - ${i}`);
+      await selectCardFromCatalog(testCard);
+      await click('[data-test-send-message-btn]');
+    }
+
+    assertMessages(assert, [
+      {
+        from: 'testuser',
+        message: 'Message - 1',
+        cards: [{ id: testCard, title: 'Hassan' }],
+      },
+      {
+        from: 'testuser',
+        message: 'Message - 2',
+        cards: [{ id: testCard, title: 'Hassan' }],
+      },
+      {
+        from: 'testuser',
+        message: 'Message - 3',
+        cards: [{ id: testCard, title: 'Hassan' }],
+      },
+    ]);
+
+    //Test the scenario where there is an update to the card
+    await click(
+      `[data-test-stack-card="${testRealmURL}index"] [data-test-cards-grid-item="${testCard}"]`,
+    );
+    await click(`[data-test-stack-card="${testCard}"] [data-test-edit-button]`);
+    await fillIn(
+      '[data-test-field="firstName"] [data-test-boxel-input]',
+      'Updated Name',
+    );
+    await click(`[data-test-stack-card="${testCard}"] [data-test-edit-button]`);
+
+    await fillIn('[data-test-message-field]', `Message with updated card`);
+    await click('[data-test-send-message-btn]');
+
+    assertMessages(assert, [
+      {
+        from: 'testuser',
+        message: 'Message - 1',
+        cards: [{ id: testCard, title: 'Updated Name Abdel-Rahman' }],
+      },
+      {
+        from: 'testuser',
+        message: 'Message - 2',
+        cards: [{ id: testCard, title: 'Updated Name Abdel-Rahman' }],
+      },
+      {
+        from: 'testuser',
+        message: 'Message - 3',
+        cards: [{ id: testCard, title: 'Updated Name Abdel-Rahman' }],
+      },
+      {
+        from: 'testuser',
+        message: 'Message with updated card',
+        cards: [{ id: testCard, title: 'Updated Name Abdel-Rahman' }],
+      },
+    ]);
+
+    // // There must be a new card fragments event because the card has been updated.
+    // events = await getRoomEvents();
+    // messageEvents = events.filter(
+    //   (e) =>
+    //     e.type === 'm.room.message' &&
+    //     e.content.msgtype === 'org.boxel.message',
+    // );
+    // cardFragmentEvents = events.filter(
+    //   (e) =>
+    //     e.type === 'm.room.message' &&
+    //     e.content.msgtype === 'org.boxel.cardFragment' &&
+    //     !e.content.data.nextFragment,
+    // );
+    // expect(messageEvents.length).toEqual(4);
+    // expect(cardFragmentEvents.length).toEqual(3);
+  });
```

Comment:
Is this meant to be commented out? It looks like it wasn’t in the original test.

Reply:
Whoops, those checks were done in the matrix suite version, but I think it is better for this acceptance test to not to assert on the inner workings and instead only validate what the user experience is.